src/tactic/converter/to_factors.lean

```diff
@@ -0,0 +1,44 @@
+/-
+Copyright (c) 2020 Alex J. Best. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Alex J. Best
+
+Factor naturals inside of a conv.
+-/
+import data.nat.prime
+import tactic.converter.interactive
+import tactic.norm_num
+
+namespace conv
+open tactic
+
+/-- `to_factors_aux` tactic factors a `nat` expression into prime factors,
+returning a proof that the original expession is product of these factors,
+expanded as a sequence of `mul`'s. The tactic does not provide proofs that
+the factors are indeed prime. -/
+meta def to_factors_aux : expr → tactic (expr × expr) :=
+λ e₁, do
+  α ← infer_type e₁,
+  c ← mk_instance_cache α,
+  match α with
```

Comment:
This condition can be clearer with ``guard $ α = `(nat)`` instead of the pattern match. It should also give a helpful error message when it's not `nat`.